jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.c

```diff
@@ -420,19 +422,23 @@ ecma_date_make_day (ecma_number_t year, /**< year value */
                     ecma_number_t date) /**< date value */
 {
   /* 1. */
-  if (!ecma_number_is_finite (year) || !ecma_number_is_finite (month) || !ecma_number_is_finite (date)
-      || fabs (year) > INT32_MAX)
+  if (!ecma_number_is_finite (year) || !ecma_number_is_finite (month) || !ecma_number_is_finite (date))
   {
     return ecma_number_make_nan ();
   }
 
   /* 2., 3., 4. */
-  int32_t y = (int32_t) (year);
   ecma_number_t m = ecma_number_trunc (month);
   ecma_number_t dt = ecma_number_trunc (date);
 
   /* 5. */
-  int32_t ym = y + (int32_t) (floor (m / 12));
+  year += (floor (m / 12));
+  if (fabs (year) > 275800)
```

Comment:
Where does this magic number come from? Please add a macro with a description for it.